c++/src/kj/compat/http.c++

```diff
@@ -2362,23 +2362,22 @@ public:
   }
 
   Promise<void> write(const void* buffer, size_t size) override {
-    if (size == 0) return kj::READY_NOW;  // can't encode zero-size chunk since it indicates EOF.
+    if (size == 0) co_return;  // can't encode zero-size chunk since it indicates EOF.
 
     auto header = kj::str(kj::hex(size), "\r\n");
     auto parts = kj::heapArray<ArrayPtr<const byte>>(3);
```

Comment:
This can be an on-stack array now.

Reply:
Also `header` could be serialized into an on-stack char array, since the size of the array cannot exceed 19 characters (including NUL terminator).